src/main/java/com/powsybl/nad/NetworkAreaDiagram.java

```diff
@@ -88,6 +92,14 @@ public void draw(Path svgFile, SvgParameters svgParameters, LayoutParameters lay
         draw(svgFile, svgParameters, layoutParameters, styleProvider, labelProvider, layoutFactory, new IntIdProvider());
     }
 
+    public Map<String, Point> layout(LayoutParameters layoutParameters) {
```

Comment:
As said in powsybl/powsybl-dev-tools#59 I think we can get rid of this `layout` methods

Reply:
They may be useful if we do not want to build the final visualization (SVG) or have an alternative visualization?

Reply:
Yes indeed! But as said in powsybl/powsybl-dev-tools#59 (let's continue the discussion here, it's more appropriate!) shouldn't we rather add this in the Layout interface then? As the return value of `Layout::run`

Reply:
Agree, you are right.

But I think then we will need to expose the graph built by the Diagram (or return a layout algorithm that has been already initialized with the right graph). So the user would see something like:

```java
Diagram d = new Diagram(...);
Graph g = d.buildGraph();
Positions p = d.layout(graph).run();
```

Or 

```java
Diagram d = new Diagram(...);
Positions p = d.layout().run();
```

Reply:
In fact you can do something like that already, similarly to what is done in `NetworkAreaDiagram::draw` method, as the graph builder is exposed. If you except the fact that `Layout::run` is not returning the positions (yet).
https://github.com/powsybl/powsybl-network-area-diagram/blob/996e2cebf7abd41e8b62551653e7345fb2bdb93c/src/main/java/com/powsybl/nad/NetworkAreaDiagram.java#L136-L138

In fact this question meets the question of what is a `Diagram` object, and of whether it is of any use to have that kind of object. Before you shared this I was more thinking of changing `NetworkAreaDiagram` class to a utility class with only static methods (like the `SingleLineDiagram` one). Indeed we could think of the diagram as a graph with calculated positions, which are within the graph in fact. But if we separate the graph from the positions it would give more sense to a diagram object.

Reply:
I think it is a good idea that the `Diagram` encapsulates how the graph has to be built. It has been already created or configured to know about the `voltageLevelFilter` and `idProvider`.
Following the same reasoning, I think it is also ok that Diagram is the owner of the layout method that will be used, but it should expose it so the user could call directly the new methods we want to add to its API. 
My proposal is:

```java
Diagram diagram = new Diagram(...)
Graph graph = diagram.buildGraph();
Layout layout = d.getLayout(layoutParameters);
layout.setInitialNodePositions(...);
layout.setNodesWithFixedPosition(...);
Map<String, Point> positions = layout.run(graph);
```

What do you think?

Reply:
Layout parameters could be passed directly to the run method of layout, like it is done now:

```java
Diagram diagram = new Diagram(...)
Graph graph = diagram.buildGraph();
Layout layout = d.getLayout();
layout.setInitialNodePositions(...);
layout.setNodesWithFixedPosition(...);
Map<String, Point> positions = layout.run(graph, layoutParameters);
```

Reply:
I agree with your vision of the Diagram object as the sum of a graph plus its positions.

I always face the dilemma of whether build a complex object that is able to do one or two things or build a "slim" object with complex methods (current draw variants on Diagram). What do you think about moving the information currently received in these draw methods to secondary constructors of the Diagram object itself?

Reply:
Just detailing why I'm not (yet) convinced by the changes you suggested. So far the diagram object does not really make sense, it is nothing more than a list of methods. In fact the constructors I wrote are a bit clumsy, as it's retaining things which would / should not be used twice. And yes, we could add parameters like the graph, the layout, the positions, so that it's really a meaningful object, but it's the opposite choice of what we did in `powsybl-single-line-diagram`.

In fact we used to have there something similar to what you suggest and we went back from this for several reasons:
- we didn't identify any usecase of calling twice draw or layout on a diagram object (other than development purposes), nor any usecase of playing with the diagram before drawing it
- just one line to call to draw a diagram, through a static method, instead of needing a tutorial needed to be able to use it
- not wondering which arguments should be in the constructor and which in the draw call, not wondering when handling a diagram whether its graph has already been build or not, whether the positions have already been computed or not